app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -7,6 +7,26 @@ def run
           results = SerializableArray.new(results)
           SearchTaskDefinition.create_output(results, output)
         end
+
+        def self.dereference_output(storage, path)
```

Comment:
I understand the need for bening able to reference things that are not just hashes and arrays. My concern with the current implementation is however, that it handles deserialization directly in the referencing. I think the `input`/`output` proxy objects might be more smart that they are right now (just plain hashes) and I'm willing to give you a coding hand in making the change in Dynflow, as it would make dynflow more usable. The dereferencing part would that became trivial. What do you think about that approach?